src/extensions/surveys.tsx

```diff
@@ -36,82 +36,154 @@ import {
 const window = _window as Window & typeof globalThis
 const document = _document as Document
 
-const handleWidget = (posthog: PostHog, survey: Survey) => {
-    const shadow = createWidgetShadow(survey)
-    const surveyStyleSheet = style(survey.appearance)
-    shadow.appendChild(Object.assign(document.createElement('style'), { innerText: surveyStyleSheet }))
-    Preact.render(<FeedbackWidget key={'feedback-survey'} posthog={posthog} survey={survey} />, shadow)
-}
+export class SurveyManager {
+    private posthog: PostHog
+    private surveysInFocus: Set<string>
+
+    constructor(posthog: PostHog) {
+        this.posthog = posthog
+        // We use a set to keep track of surveys in focus to prevent multiple surveys from showing at the same time
+        // This is important for correctly displaying popover surveys with a delay, where we want to show them
+        // in order of their delay, rather than evaluate them all at once.
+        // NB: This set should only ever have 0 or 1 items in it at a time.
+        this.surveysInFocus = new Set<string>()
```

Comment:
Aligned, and I love that idea.

Reply:
Done, let me know what you think of the error message (I added it to the `addSurveyToFocus` method)

Reply:
error looks good! Although it's still a set 😅  - sorry if this wasn't clear in my earlier message, I was suggesting making it a string | null, instead of a set, since to me there doesn't seem to be a need for a set here

Reply:
(which is what creates the confusion for me, since if its a set, I somewhat expect to see it handling more than one item!)